### PR TITLE
setup.py: use symlinks to strip .py from scripts

### PR DESCRIPTION
**Pull request by floppym**
Overriding distutils/setuptools classes is fragile and error-prone.

Fixes: https://github.com/mesonbuild/meson/issues/1677

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -25,33 +25,18 @@
...
-
-        # We want the files to be installed without a suffix on Unix
-        for infile in self.get_inputs():
-            in_stripped = infile[:-3] if infile.endswith('.py') else infile
-            outfile = os.path.join(self.install_dir, in_stripped)
-            # NOTE: Mode is preserved by default
-            copy_file(infile, outfile, dry_run=self.dry_run)
-            self.outfiles.append(outfile)
+if sys.platform == 'win32':
+    scripts=['meson.py',
```
Comment (ignatenkobrain) [inline setup.py]: __[flake8]__
* [E225] missing whitespace around operator

File: setup.py [hunk 2]
```diff
@@ -25,33 +25,18 @@
...
+             'mesonconf.py',
+             'mesontest.py',
+             'mesonintrospect.py',
+             'wraptool.py']
+else:
+    scripts=['scripts/meson',
```
Comment (ignatenkobrain) [inline setup.py]: __[flake8]__
* [E225] missing whitespace around operator

**Comment by jpakkane** [comment]
This message is posted to every outstanding merge request.

We have transitioned from Github wiki to our new documentation system that generates all documentation directly from the Git repo. This means that from now on every merge request must come with corresponding documentation changes.

If your MR requires documentation changes, do the necessary changes to files in `docs/markdown`.

If your MR adds a new feature, add a description to the release notes in `docs/markdown/Release-notes-for-0.41.0.md`.

If this commit is just a bugfix with no functional changes, you don't need to do anything.

Thank you

**Comment by jpakkane** [comment]
I'd prefer not to have symlinks in the repo. They break things in subtle ways on Windows.

**Comment by nirbheek** [comment]
We probably want to call `distutils.command.build_scripts.copy_scripts()` and then move the files from `meson.py -> meson` etc?

**Comment by floppym** [comment]
I created #1705 as an alternative solution.

**Pull request closed without merging**